google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/impl/PubSubApi.scala

```diff
@@ -164,7 +164,7 @@ private[pubsub] trait PubSubApi {
       .mapAsync(1) {
         case Success(response) =>
           response.status match {
-            case StatusCodes.Success(_) =>
+            case StatusCodes.Success(_) if response.entity.contentType == ContentTypes.`application/json` =>
```

Comment:
Maybe also include the contentType in the "Unexpected response" exception below, to help debugging when the response is rejected because the content-type is wrong?